src/worker/tasks/sendTransactionWorker.ts

```diff
@@ -352,37 +350,41 @@ const _sendTransaction = async (
       await account.sendTransaction(populatedTransaction);
     transactionHash = sendTransactionResult.transactionHash;
   } catch (error: unknown) {
-    // If the nonce is already seen onchain (nonce too low) or in mempool (replacement underpriced),
-    // correct the DB nonce.
-    if (isNonceAlreadyUsedError(error) || isReplacementGasFeeTooLow(error)) {
-      const result = await syncLatestNonceFromOnchainIfHigher(chainId, from);
-      job.log(`Re-synced nonce: ${result}`);
-    } else {
-      // Otherwise this nonce is not used yet. Recycle it to be used by a future transaction.
-      job.log(`Recycling nonce: ${nonce}`);
-      await recycleNonce(chainId, from, nonce);
-    }
-
-    // Do not retry errors that are expected to be rejected by RPC again.
     if (isInsufficientFundsError(error)) {
-      const { name, nativeCurrency } = await getChainMetadata(chain);
+      // Insufficient funds. Do not retry
       const { gas, value = 0n } = populatedTransaction;
+      const { name, nativeCurrency } = await getChainMetadata(chain);
+
+      // This and other pending transactions will fail.
+      // Reset the nonce state for this wallet. The first transaction after the wallet is funded will resync the nonce.
+      if (value === 0n) {
+        await deleteNoncesForBackendWallets([{ chainId, walletAddress: from }]);
+      }
```

Comment:
This is the main change. If a transaction received an "insufficient funds" error from RPC and no value was provided, this wallet must be low on gas. Reset the nonce so the onchain nonce doesn't remain out of sync.